tests/foreman/api/test_permission.py

```diff
@@ -41,17 +57,22 @@ def test_search_by_name(self, permission_name):
         is the one searched for.
 
         """
-        if (
-                permission_name == 'power_compute_resources_vms' and
-                bz_bug_is_open(1198731)):
-            self.skipTest('BZ 1198731 is open.')
-        result = entities.Permission(name=permission_name).search()
-        self.assertEqual(len(result), 1, permission_name)
-        self.assertEqual(permission_name, result[0]['name'])
+        failures = {}
+        for permission_name in self.permission_names:
+            result = entities.Permission(name=permission_name).search()
+            length = len(result)
+            name = result[0]['name'] if length > 0 else None
+            if length != 1 or permission_name != name:
+                failures[permission_name] = {
+                    'length': len(result),
+                    'returned_name': name,
+                }
+
+        if failures:
+            self.fail(json.dumps(failures, indent=True, sort_keys=True))
```

Comment:
This is great. There are a lot of permissions that can be searched for, and collecting all failures before failing will allow for much nicer test results than simply failing as soon as a single test failure is found.